src/wp-includes/functions.php

```diff
@@ -7635,3 +7633,85 @@ function is_php_version_compatible( $required ) {
 function wp_fuzzy_number_match( $expected, $actual, $precision = 1 ) {
 	return abs( (float) $expected - (float) $actual ) <= $precision;
 }
+
+/**
+ * Formats attributes string from associative array $attributes to be placed insed `<script>` tag.
+ *
+ * Automatically injects type attribute if needed.
+ * Used by wp_print_script_loader_tag() and wp_print_inline_script_tag().
+ *
+ * @param array     $attributes     Optional. `<script>` tag attributes.
+ * @param bool      $double_quotes  Optional. Uses double quotes `"` to wrap attribute values if true, single quotes `'` otherwise.
+ * @return string  Inline JavaScript code wrapped around <script> tags if $echo is true, null otherwise.
+ */
+function wp_sanitize_script_attributes( $attributes = array() ) {
+	if ( ! array_key_exists( 'type', $attributes ) && ! is_admin() && ! current_theme_supports( 'html5', 'script' ) ) {
+		$attributes['type'] = 'text/javascript';
+	}
+	/**
+	 * Filters script attributes.
+	 *
+	 * @param string $attributes    Associative array representing script attributes.
+	 */
+	$attributes = apply_filters( 'wp_script_attributes', $attributes );
+
+	$attributes_string = '';
+	// Only the attribute name is added to $attributes_string for entries with a boolean value, and that are true.
+	foreach ( $attributes as $attribute_name => $attribute_value ) {
+		if ( is_bool( $attribute_value ) ) {
+			if ( $attribute_value && '' !== $attribute_name ) {
+				$attributes_string .= ' ' . $attribute_name;
+			}
+		} else {
+			$attributes_string .= sprintf( ' %s="%s"', $attribute_name, esc_attr( $attribute_value ) );
+		}
+	}
+
+	return $attributes_string;
+}
+
+/**
+ * Wraps inline JavaScript in `<script>` tags.
+ *
+ * Enables to inject attributes in the `<script>` tag via the `wp_script_attributes` filter.
+ * Automatically injects type attribute if needed.
+ *
+ * @param array     $attributes     `<script>` tag attributes.
+ * @param bool      $double_quotes  Optional. Uses double quotes `"` to wrap attribute values if true, single quotes `'` otherwise.
+ * @param bool      $echo           Optional. Prints to the page if true, returns the value otherwise.
+ * @param string    $content        Optional. Code to be wrapped around `<script>` tags.
+ * @return string|null  Inline JavaScript code wrapped around `<script>` tags if $echo is true, null otherwise.
+ */
+function wp_print_script_loader_tag( $attributes, $echo = true ) {
```

Comment:
When would someone **not** want to `echo` this? The function's name is `wp_print_script_loader_tag `, not `wp_get_script_loader_tag`, so let's remove the `echo` parameter or introduce `wp_get_script_loader_tag` if really needed.

Reply:
See above.
Example: https://github.com/enricocarraro/wordpress-develop/blob/cda80692b51cd2d9705ba8543f4ab36b14518b20/src/wp-includes/class.wp-scripts.php#L394-L401

Reply:
Then I'd prefer `wp_get_script_loader_tag` over this. The `$echo` thing is a bit of an anti-pattern.

Reply:
Sure, no problem.